src/charm.py

```diff
@@ -583,8 +583,8 @@ def _check_and_update_dependencies(self) -> bool:
 
             self.unit.status = MaintenanceStatus("Flushing runners due to updated deps")
 
+            runner_manager.flush(wait_repo_check=True)
```

Comment:
Do we really want to flush busy runners when the runner binary has been updated?

Reply:
Probably it is fine not to

Reply:
Changed to not flush busy runners.